spec/system/admin/facility_orders_search_spec.rb

```diff
@@ -3,53 +3,191 @@
 
-  let!(:completed_orders) do
-    accounts.map do |account|
-      [create(:complete_order, product: item, account: account),
-       create(:complete_order, product: item2, account: account)]
-    end.flatten
+  before do
+    login_as director
   end
 
-  before do
-    completed_orders.each do |order|
-      order.order_details.first.update(price_policy: nil)
+  context "same facility orders" do
+    let!(:orders) do
+      accounts.map do |account|
+        [create(:purchased_order, product: item, account: account),
+         create(:purchased_order, product: item2, account: account)]
+      end.flatten
+    end
+    let!(:completed_orders) do
+      accounts.map do |account|
+        [create(:complete_order, product: item, account: account),
+         create(:complete_order, product: item2, account: account)]
+      end.flatten
+    end
+
+    before do
+      completed_orders.each do |order|
+        order.order_details.first.update(price_policy: nil)
+      end
+    end
+
+    context "new and in process orders tab" do
+      it "can do a basic search" do
+        visit facility_orders_path(facility)
+
+        select item.to_s, from: "Products"
+        click_button "Filter"
+        expect(page).to have_css(".order-detail-description", text: item.name, count: 2)
+        expect(page).not_to have_css(".order-detail-description", text: item2.name)
+      end
     end
-  end
 
-  context "new and in process orders tab" do
-    it "can do a basic search" do
-      login_as director
-      visit facility_orders_path(facility)
+    context "problem orders tab" do
+      it "can do a basic search" do
+        visit show_problems_facility_orders_path(facility)
 
-      select item.to_s, from: "Products"
-      click_button "Filter"
-      expect(page).to have_css(".order-detail-description", text: item.name, count: 2)
-      expect(page).not_to have_css(".order-detail-description", text: item2.name)
+        select item.to_s, from: "Products"
+        click_button "Filter"
+        expect(page).to have_css("td", text: item.name, count: 2)
+        expect(page).not_to have_css("td", text: item2.name)
+      end
     end
   end
 
-  context "problem orders tab" do
-    it "can do a basic search" do
-      login_as director
-      visit show_problems_facility_orders_path(facility)
+  context "cross-core orders", :js, feature_setting: { cross_core_order_view: true } do
+    # cross_core_order_originating_facility is a purchased order that originates in Facility (item)
+    # facility2_item is ordered and that new order is added to cross_core_project
+
+    # cross_core_order_originating_facility2 is a purchased order that originates in Facility2 (facility2_item)
+    # item is ordered and that new order is added to cross_core_project2
+
+    # cross_core_project has cross_core_order_originating_facility order and another one for Facility2
+    # cross_core_project2 has cross_core_order_originating_facility2 order and another one for Facility
```

Comment:
This section was a little confusing to me - if there's useful context here for the future, I'd move it closer to the code it's describing.  The relations are clear from looking at the `let` statements, so code comments should tell us "What are we testing here?" or "why did we set up the test this way?"

Reply:
This started as notes for myself, as I was a bit confused while writing the spec. I updated the comments to only have the useful parts. Let me know what you think